lib/linter/linter.js

```diff
@@ -386,12 +447,13 @@ function createDisableDirectives({ type, value, justification, node }, ruleMappe
  * where reporting is disabled or enabled and merges them with reporting config.
  * @param {SourceCode} sourceCode The SourceCode object to get comments from.
  * @param {function(string): {create: Function}} ruleMapper A map from rule IDs to defined rules
+ * @param {StringSeverity} reportUnusedInlineConfigs Whether to report on unused inline config comments.
  * @param {string|null} warnInlineConfig If a string then it should warn directive comments as disabled. The string value is the config name what the setting came from.
  * @param {ConfigData} config Provided config.
  * @returns {{configuredRules: Object, enabledGlobals: {value:string,comment:Token}[], exportedVariables: Object, problems: LintMessage[], disableDirectives: DisableDirective[]}}
  * A collection of the directive comments that were found, along with any problems that occurred when parsing
  */
-function getDirectiveComments(sourceCode, ruleMapper, warnInlineConfig, config) {
+function getDirectiveComments(sourceCode, ruleMapper, reportUnusedInlineConfigs, warnInlineConfig, config) {
```

Comment:
This function is used in eslintrc mode only, so there's no need for any changes in it